src/Microsoft.AspNetCore.OData/Batch/ODataBatchMiddleware.cs

```diff
@@ -34,13 +34,21 @@ public ODataBatchMiddleware(RequestDelegate next)
         /// <returns>A task that can be awaited.</returns>
         public async Task Invoke(HttpContext context)
         {
+            var request = context.Request;
+            var isPreFlight = request.Method.Equals("Options", StringComparison.OrdinalIgnoreCase);
+            if (isPreFlight)
+            {
+                // if the request is a preflight request let the next handler be called as trying to handle the request will result in preflight requests failing.
+                await this.next(context);
```

Comment:
>await [](start = 16, length = 5)

in this scenario, you can probably just return this.next(context) to avoid the final await.  see [https://blog.stephencleary.com/2016/12/eliding-async-await.html](https://blog.stephencleary.com/2016/12/eliding-async-await.html) on eliding the final await.